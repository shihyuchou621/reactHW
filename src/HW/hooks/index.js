import HW1 from './HW1';
import HW2 from './HW2';
import HW3 from './HW3';
import HW4 from './HW4';
import HW5 from './HW5';
import HW6 from './HW6';
import HW7 from './HW7';
import HW8 from './HW8';
import HW9 from './HW9';
import HW10 from './HW10';
import HW11 from './HW11';
import HW12 from './HW12';
import HW13 from './HW13';
import HW14 from './HW14';
import HW15 from './HW15';
import HW16 from './HW16';
import HW17 from './HW17';
import HW18 from './HW18';
import HW19 from './HW19';
import HW20 from './HW20';
import HW21 from './HW21';
import HW22 from './HW22';
import HW23 from './HW23';
import HW25 from './HW25';
import HW26 from './HW26';
import HW27 from './HW27';

export default {
  HW1,
  HW2,
  HW3,
  HW4,
  HW5,
  HW6,
  HW7,
  HW8,
  HW9,
  HW10,
  HW11,
  HW12,
  HW13,
  HW14,
  HW15,
  HW16,
  HW17,
  HW18,
  HW19,
  HW20,
  HW21,
  HW22,
  HW23,
  HW25,
  HW26,
  HW27,
};