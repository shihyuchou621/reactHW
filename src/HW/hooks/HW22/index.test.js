import React from 'react';
import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from '.';

afterEach(cleanup);

test('should have correct subtotal and total', () => {
  const { container, getByTestId } = render(<HW />);
  userEvent.type(container.querySelector('input[name="橘子"]'), '3{enter}');
  userEvent.type(container.querySelector('input[name="蓮霧"]'), '1{enter}');
  userEvent.type(container.querySelector('input[name="哈蜜瓜"]'), '1{enter}');
  userEvent.type(container.querySelector('input[name="蘋果"]'), '2{enter}');
  userEvent.type(container.querySelector('input[name="葡萄柚"]'), '5{enter}');

  // 上面那行應該就可以改成這樣... 有空再改改看
  // userEvent.type(container.querySelector('input[name="葡萄柚"]'),
  //   "5"
  // );

  expect(container).toMatchSnapshot();
  expect(getByTestId("subtotal橘子")).toHaveTextContent("橘子 x 3 = $60");
  expect(getByTestId("subtotal蓮霧")).toHaveTextContent("蓮霧 x 1 = $30");
  expect(getByTestId("subtotal哈蜜瓜")).toHaveTextContent("哈蜜瓜 x 1 = $40");
  expect(getByTestId("subtotal蘋果")).toHaveTextContent("蘋果 x 2 = $100");
  expect(getByTestId("subtotal葡萄柚")).toHaveTextContent("葡萄柚 x 5 = $75");
  expect(getByTestId("total")).toHaveTextContent("總計 $305");

});

test('should minus and clear correctly', () => {
  const { container, getByTestId, getByText } = render(<HW />);
  userEvent.type(container.querySelector('input[name="橘子"]'), '3{enter}');
  userEvent.type(container.querySelector('input[name="蓮霧"]'), '2{enter}');
  userEvent.click(getByTestId("橘子-1"));
  expect(getByTestId("subtotal橘子")).toHaveTextContent("橘子 x 2 = $40");

  userEvent.click(getByTestId("蓮霧clear"));
  expect(container).toMatchSnapshot();
});

/**
   * 我剛剛看到一個東西，好像可以讓你的 change 更方便
   * https://github.com/testing-library/user-event#typeelement-text-options
   *
   * import userEvent from '@testing-library/user-event';
   * ...
   * userEvent.type(screen.getByXXX('xxxx'), '要輸入的內容{enter}');
   * 這樣你就不用打 change 然後還要放 target.value.....
   * 喏?
  */

/** 男子~~~漢~~~~~
   * 行動快速像那江河湍急~
   * 男子~~~漢~~~~~
   * 破壞力像那風暴無情~
  */