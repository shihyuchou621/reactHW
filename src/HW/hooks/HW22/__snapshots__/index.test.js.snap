// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have correct subtotal and total 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="productArea"
    >
      <div
        class="product"
      >
        <div>
          橘子
        </div>
        <div>
          $
          20
        </div>
        <form
          data-testid="form"
          name="橘子"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="橘子"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          蓮霧
        </div>
        <div>
          $
          30
        </div>
        <form
          data-testid="form"
          name="蓮霧"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="蓮霧"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          哈蜜瓜
        </div>
        <div>
          $
          40
        </div>
        <form
          data-testid="form"
          name="哈蜜瓜"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="哈蜜瓜"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          蘋果
        </div>
        <div>
          $
          50
        </div>
        <form
          data-testid="form"
          name="蘋果"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="蘋果"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          葡萄柚
        </div>
        <div>
          $
          15
        </div>
        <form
          data-testid="form"
          name="葡萄柚"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="葡萄柚"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
    </div>
    <div
      class="cartArea"
    >
      <div
        class="productInCart"
        data-testid="subtotal橘子"
      >
        <span
          class="subtotal"
        >
          橘子
           x 
          3
           = $
          60
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="橘子clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="橘子-1"
        >
          -1
        </button>
      </div>
      <div
        class="productInCart"
        data-testid="subtotal蓮霧"
      >
        <span
          class="subtotal"
        >
          蓮霧
           x 
          1
           = $
          30
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="蓮霧clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="蓮霧-1"
        >
          -1
        </button>
      </div>
      <div
        class="productInCart"
        data-testid="subtotal哈蜜瓜"
      >
        <span
          class="subtotal"
        >
          哈蜜瓜
           x 
          1
           = $
          40
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="哈蜜瓜clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="哈蜜瓜-1"
        >
          -1
        </button>
      </div>
      <div
        class="productInCart"
        data-testid="subtotal蘋果"
      >
        <span
          class="subtotal"
        >
          蘋果
           x 
          2
           = $
          100
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="蘋果clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="蘋果-1"
        >
          -1
        </button>
      </div>
      <div
        class="productInCart"
        data-testid="subtotal葡萄柚"
      >
        <span
          class="subtotal"
        >
          葡萄柚
           x 
          5
           = $
          75
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="葡萄柚clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="葡萄柚-1"
        >
          -1
        </button>
      </div>
      <div>
        <div>
          ===============
        </div>
        <div
          data-testid="total"
        >
          總計 $
          305
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should minus and clear correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="productArea"
    >
      <div
        class="product"
      >
        <div>
          橘子
        </div>
        <div>
          $
          20
        </div>
        <form
          data-testid="form"
          name="橘子"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="橘子"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          蓮霧
        </div>
        <div>
          $
          30
        </div>
        <form
          data-testid="form"
          name="蓮霧"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="蓮霧"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          哈蜜瓜
        </div>
        <div>
          $
          40
        </div>
        <form
          data-testid="form"
          name="哈蜜瓜"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="哈蜜瓜"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          蘋果
        </div>
        <div>
          $
          50
        </div>
        <form
          data-testid="form"
          name="蘋果"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="蘋果"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
      <div
        class="product"
      >
        <div>
          葡萄柚
        </div>
        <div>
          $
          15
        </div>
        <form
          data-testid="form"
          name="葡萄柚"
        >
          <input
            class="input"
            data-testid="Product"
            max="99"
            min="0"
            name="葡萄柚"
            placeholder="數量"
            type="number"
          />
          <button
            class="btn btn-outline-primary btn-sm"
          >
            加入購物車
          </button>
        </form>
      </div>
    </div>
    <div
      class="cartArea"
    >
      <div
        class="productInCart"
        data-testid="subtotal橘子"
      >
        <span
          class="subtotal"
        >
          橘子
           x 
          2
           = $
          40
        </span>
        <button
          class="buttonInCart  btn btn-danger btn-sm"
          data-testid="橘子clear"
        >
          clear
        </button>
        <button
          class="buttonInCart  btn btn-warning btn-sm"
          data-testid="橘子-1"
        >
          -1
        </button>
      </div>
      <div>
        <div>
          ===============
        </div>
        <div
          data-testid="total"
        >
          總計 $
          40
        </div>
      </div>
    </div>
  </div>
</div>
`;
